include/oneapi/dpl/internal/exclusive_scan_by_segment_impl.h

```diff
@@ -115,12 +114,11 @@ exclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIter
     typedef typename ::std::iterator_traits<OutputIterator>::value_type OutputType;
     typedef typename ::std::iterator_traits<InputIterator2>::value_type ValueType;
     typedef unsigned int FlagType;
-    typedef ::std::decay_t<Policy> policy_type;
 
     InputIterator2 last2 = first2 + n;
 
     // compute head flags
-    internal::__buffer<policy_type, FlagType> _flags(policy, n);
+    oneapi::dpl::__par_backend_hetero::__internal::__buffer<Policy, FlagType> _flags(policy, n);
     {
         auto flag_buf = _flags.get_buffer();
         auto flags = flag_buf.get_host_access(sycl::read_write);
```

Comment:
I am surprised with `get_host_access` here and on line 135. The all computation must be done on a device, we must not getting the host access. Yes, it is a different(from this PR) story... But, lets don't  forget to fix it. Probably, make sense to create a reference issue.

Reply:
Issue created: https://github.com/oneapi-src/oneDPL/issues/1199